lib/vrl/stdlib/src/flatten.rs

```diff
@@ -3,13 +3,20 @@ use std::collections::btree_map;
 use ::value::Value;
 use vrl::prelude::*;
 
-fn flatten(value: Value) -> Resolved {
+static DEFAULT_SEPARATOR: &str = ".";
+
+fn flatten(value: Value, separator: Option<Value>) -> Resolved {
+    let separator = match separator {
+        None => DEFAULT_SEPARATOR.to_string(),
+        Some(separator) => separator.try_bytes_utf8_lossy()?.to_string(),
+    };
```

Comment:
Could `separator` just be a `&str` rather than an owned `String`?  Below it is just used in a `format!()` so it seems like it could be. It's a small thing, but this is executed on each function call so eliminating the allocation could improve performance.

Reply:
Would it be more idiomatic to do something like `if let Some(separator)...`?